include/mbedtls/platform_util.h

```diff
@@ -55,6 +66,38 @@ extern "C" {
  */
 void mbedtls_platform_zeroize( void *buf, size_t len );
 
+#if defined(MBEDTLS_HAVE_TIME_DATE)
+/**
+ * \brief       Thread safe implementation of gmtime()
+ *
+ *              The function is an abstraction that when called behaves similar
+ *              to the gmtime() function from the C standard, but is thread
+ *              safe.
+ *
+ *              Mbed TLS will try to identify the underlying platform and
+ *              configure an appropriate underlying implementation (e.g.
+ *              gmtime_r() for POSIX and gmtime_s() for Windows). If this is
+ *              not possible, then gmtime() will be used. In this case, calls
```

Comment:
This means that *any* call to `gmtime()` outside the library is required to use this mbedtls mutex. I don't like this too much, but I understand the limitation in this part.
Isn't there a way to add a setup for an external mutex in this case, perhaps in the platform context?

Reply:
I think the present version, plus the documentation that other calls to `gmtime()` should be synchronized with the same mutex, is the best we can do based if we only have `gmtime()`.

> Isn't there a way to add a setup for an external mutex in this case, perhaps in the platform context?

I don't quite understand this - the threading mutex used for the default gmtime implementation is public, so application code may use it. Could you perhaps rephrase your point?

Reply:
We may be the only users of `gmtime()` or it may be on a single threaded platform. Both seem to be the case for Mbed Cloud Client and Mbed OS.

Reply:
@hanno-arm let me rephrase, I prefer to have a way for the mutex to be defined by the application, and given to use as a parameter.
The application may use it, but I don't think it's reasonable for the application to be using an `mbedtls_platform_gmtime_mutex` for parts of the application that is not related to Mbed TLS.

Reply:
Thanks, that's what I thought.

>The mutex interface used by Mbed TLS is well documented and there is no reason why careful users wouldn't be able to use it.

The only reason I can find is consistency and design in the whole application, using Mbed TLS related naming in a non mbed TLs module. but that's probably a necessity.

Reply:
@RonEld Yes, I think so. And if users dislike using our mutexes, they can still provide their custom `gmtime_r()` implementation building on whatever mutex implementation they prefer.